common/scala/src/main/scala/org/apache/openwhisk/core/entity/WhiskAction.scala

```diff
@@ -548,8 +632,12 @@ object WhiskAction extends DocumentFactory[WhiskAction] with WhiskEntityQueries[
       wp flatMap { resolvedPkg =>
         // fully resolved name for the action
         val fqnAction = resolvedPkg.fullyQualifiedName(withVersion = false).add(actionName)
+        val action = WhiskActionVersionList.get(fqnAction, entityStore).flatMap { result =>
```

Comment:
I am still unsure about having two separate calls to fetch a certain version of the action.
I haven't tried this but there are some tricks to sort data like https://stackoverflow.com/a/41959344

If we can sort the data using the views we may have something like this with one list query:

```
actionA@0.0.3
actionA@0.0.2
actionA@0.0.1
```

Then we might be able to get the latest action metadata with the `limit=1` parameter.

```
actionA@0.0.3
```

Also, if a DocId includes the version information, we can directly fetch a certain version even if it is old.

Reply:
It is still important to provide a feature in a backward-compatible way, if this is designed in this way for backward compatibility, I am inclined not to let the design be changed because of the backward compatibility. Instead, we can provide other options such as a feature flag with a way to easily migrate existing data.

I believe the semantics or behavior of the feature should not be affected because of the backward compatibility.

Reply:
It might be hard to sort since you'll have to follow semver versus just being able to do a lexicographical sort. It should be able to be done though in the view definition I think? 

But yea keep in mind:
actionA@0.0.21
actionA@0.0.9

might not get sorted correctly if you don't have semver parsing in your sort

Reply:
hm, maybe we can try use view key like `[ "whisk.system/hello", 0, 0, 1 ]`? I will check it

Reply:
seems that it cannot reuse many current codes if do this way